pymc_experimental/distributions/timeseries.py

```diff
@@ -0,0 +1,198 @@
+        Matrix of transition probabilities between states. Rows must sum to 1.
+        One of P or P_logits must be provided.
+    P_logit: tensor, Optional
+        Matrix of tranisiton logits. Converted to probabilities via Softmax activation.
+        One of P or P_logits must be provided.
+    steps: tensor
+        Length of the markov chain
+    x0: tensor or RandomVariable
+        Intial state of the system. If tensor, treated as deterministic.
+    """
+
+    rv_type = DiscreteMarkovChainRV
+
+    def __new__(cls, *args, steps, **kwargs):
+        # TODO: Allow steps to be None and infer chain length from shape?
+        # TODO: Dims breaks the RV
+
+        # Subtract 1 step to account for x0 given, better match user expectation of
+        # len(markov_chain) = steps
+        steps -= 1
+
+        steps = get_support_shape_1d(
+            support_shape=steps,
+            shape=None,
+            dims=kwargs.get('dims', None),
+            observed=kwargs.get('observed', None),
+            support_shape_offset=1
+        )
+
+        return super().__new__(cls, *args, steps=steps, **kwargs)
+
+    @classmethod
+    def dist(cls, P=None, logit_P=None, steps=None, x0=None, **kwargs):
+        steps = get_support_shape_1d(
+            support_shape=steps, shape=kwargs.get('shape', None), support_shape_offset=1
+        )
+        if steps is None:
+            raise ValueError("Must specify steps or shape parameter")
+        if P is None and logit_P is None:
+            raise ValueError('Must specify P or logit_P parameter')
+        if P is not None and logit_P is not None:
+            raise ValueError('Must specify only one of either P or logit_P parameter')
+
+        if logit_P is not None:
+            P = pm.math.softmax(logit_P, axis=1)
+        P = at.as_tensor_variable(P)
+        validate_transition_matrix(P)
+
+        # TODO: Can this eval be avoided?
+        n_states = P.shape[0].eval()
+
+        if not isinstance(x0, TensorVariable):
+            x0 = at.as_tensor_variable(x0).astype(intX)
+
+            # TODO: Can this eval be avoided?
+            if not at.all(at.lt(x0, n_states - 1)).eval():
+                raise ValueError('At least one initial state is larger than the number of states in the Markov Chain')
+
+        elif not isinstance(x0.owner.op, Discrete):
+            raise ValueError('x0 must be a discrete distribution')
+
+        else:
+            x0_probs = x0.owner.inputs[-1].eval()
+            n_cats = 1 if x0_probs.ndim == 0 else len(x0_probs)
+
+            if not n_cats <= n_states:
+                raise ValueError(
+                    'x0 has support over a range of values larger than the number of states in the Markov Chain')
+
+        return super().dist([P, logit_P, steps, x0], **kwargs)
+
+    @classmethod
+    def rv_op(cls, P, logit_P, steps, x0, size=None):
+        if size is not None:
+            batch_size = size
+        else:
+            batch_size = at.broadcast_shape(x0)
```

Comment:
This broadcast doesn't do anything

Reply:
Yes, I copied it from GARCH11, but then I didn't know what to do with it. I guess it's supposed to ensure that if i pass a vector to the parameters, the shape of those parameters will be added as a batch_dim? I'll figure out how to make sure this matches the other time series distributions in handling that case.

Reply:
This was the source of the issue with `dims`, it has been resolved, see my comment above.

Reply:
> Yes, I copied it from GARCH11, but then I didn't know what to do with it. I guess it's supposed to ensure that if i pass a vector to the parameters, the shape of those parameters will be added as a batch_dim? 

When you don't specify size explicit, it's implied by the parameters shape. If there's more than one parameter, it's determined by the broadcasted batched dimensions. When you have a single parameter there's no need to broadcast.